internal/controllers/machine/machine_controller.go

```diff
@@ -639,16 +726,10 @@ func (r *Reconciler) drainNode(ctx context.Context, cluster *clusterv1.Cluster,
 		GracePeriodSeconds:  -1,
 		// If a pod is not evicted in 20 seconds, retry the eviction next time the
 		// machine gets reconciled again (to allow other machines to be reconciled).
-		Timeout: 20 * time.Second,
-		OnPodDeletedOrEvicted: func(pod *corev1.Pod, usingEviction bool) {
-			verbStr := "Deleted"
-			if usingEviction {
-				verbStr = "Evicted"
-			}
-			log.Info(fmt.Sprintf("%s pod from Node", verbStr),
-				"Pod", klog.KObj(pod))
-		},
-		Out: writer{log.Info},
+		Timeout:                         20 * time.Second,
+		OnPodDeletionOrEvictionFinished: input.OnPodDeletionOrEvictionFinished,
+		OnPodDeletionOrEvictionStarted:  input.OnPodDeletionOrEvictionStarted,
```

Comment:
Just for my understanding:
- at every reconcile the controller runs drain for 20 seconds
- during those 20 seconds the two event handlers above are called
- each call updated the condition on the machine object - in memory -
- after 20 seconds drain times out (or completes), the reconcile loop finishes, and we update the machine -in etcd 

is that correct?

Reply:
yes, that is correct.